### ISSUE TITLE
Not allowed to set certain flags for particular tenant

### ISSUE THREAD

**Issue by scottbartell** [auth0/auth0-deploy-cli#270]
### Description

I am experiencing the same issue described here: https://community.auth0.com/t/not-allowed-to-set-certain-flags-for-particular-tenant/45926 

Basically I'm trying to create a new tenant for a dev environment, so I exported the config from my existing staging tenant, which results in a config file that includes:

```yaml
tenant:
  flags:
    ...
    include_email_in_reset_pwd_redirect: true
    include_email_in_verify_email_redirect: true
```

The problem is when I go to deploy this config to the newly created dev tenant I get the following error:
```
...
2020-08-06T19:15:26.945Z - error: Problem running command deploy during stage processChanges when processing type tenant
2020-08-06T19:15:26.945Z - error: You are not allowed to set flag 'include_email_in_reset_pwd_redirect' for this tenant.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

Same deal for `include_email_in_verify_email_redirect` if I remove the `include_email_in_reset_pwd_redirect` flag from the `tenant.yaml` file:
```
...
2020-08-06T19:08:02.620Z - error: You are not allowed to set flag 'include_email_in_reset_pwd_redirect' for this tenant.
...
```

### Environment

- **Version of this library used:** 5.1.3

**Comment by ross30152** [comment]
I am experiencing the exact same issue.  My a0deploy version is 5.1.2.

**Comment by tyler-cloud-elements** [comment]
I'm also experiencing the same issue for auth0deploy version `4.3.1`

**Comment by iousin** [comment]
Hi, 
I'm also on the same boat, please advise, getting exact same error as described above. I'm using version 5.1.5

Thanks.

**Comment by wirjo** [comment]
+1 on the same boat!

**Comment by michaelmcmillan** [comment]
Same here.

**Comment by Climber22** [comment]
Same here.

**Comment by nesaplay** [comment]
+1 looking at the same error

**Comment by TehNrd** [comment]
I am also seeing this and I opened a case with Auth0, #00466927.

This is working fine on 3 of our tenants but then failing on one. The one it is of of the new domains with a region in it like `.us`, I wonder if that is related.

Documentation about what this flag does is not super clear:
```
Whether the reset_email and verify_email templates should include the user's email address as the email parameter in the returnUrl (true) or whether no email address should be included in the redirect (false). Defaults to true.
```

...okay, email in the URL, but what is the impact of turning this off? Does the user experience change for the end user?

**Comment by wavemechanics** [comment]
I see it on versions 4.2.2 and 5.2.1.

**Comment by TehNrd** [comment]
Response from support is:

> Newer tenants do not have the flag present...it was disabled by default not because of any specific security issue that I am aware of, I think it was because we are trying to limit the amount of user information that we might potentially leak from our endpoints.

So these flags can't be set on newer tenants. It would be nice if the API just ignored these values, or this tool automatically removed them when not necessary, but not sure how to determine that at run time.

**Comment by conatus** [comment]
Also caught by this bug on a new tenant. Some sort of graceful fallback would be good.

Thanks so much!

**Comment by shushen** [comment]
Hi there!

These flags are being deprecated and may not be available for update for all tenants.

From the perspective of `auth0-deploy-cli`, it is meant to be a configuration as code tool, thus silently ignoring a not-applicable flag is not a good strategy. Throwing an error and make these deprecated flags visible could help keeping a valid configuration.

On the other hand, we are continuing improving the way we communicate deprecation and hopefully can avoid confusions in future.

I'm closing this issue as `auth0-deploy-cli` is behaving as expected.

Thanks,
Shu

**Comment by willvedd** [comment]
Hi folks, know this is coming a bit late but wanted let you all know this was fixed with [v7.14.0](https://github.com/auth0/auth0-deploy-cli/releases/tag/v7.14.0). Appreciate the patience here!
